src/date-range-input/date-range-input.tsx

```diff
@@ -273,6 +277,14 @@ export const DateRangeInput = ({
             onChange?.(val, selectedEnd);
             return;
         }
+
+        if (isFixedRangeSelection) {
```

Comment:
noticed some weird behaviour due to reusing the existing range logic
- select start date in Nov, then select start date in Dec -> end input gets focused
- clearing the start date does not clear the end date

it might be easier to explicitly split up the logic in `handleCalendarSelect` instead of mixing range and fixed range handling